### PR TITLE
Add support for v1 config.

### LINKED ISSUE #1217: Migrate "v2" config to "v1" in Mixer fitler config
Currently,  it is using "v2" which is confusing.   Decided to migrate it back to "v1" with following steps

1) change proxy to support both "v1" and "v2" fields.  if this is version x
2) After that is widely deployed,  change Pilot to "s/v2/v1", also release node that side-car proxy has to be at lease version x.  Please fiile an issue in istio.
3) after a while, remove "v2" in the proxy.

### PR DESCRIPTION
**Pull request by JimmyCYJ**
**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1217 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: src/envoy/utils/config.cc [hunk 1]
```diff
@@ -28,10 +28,31 @@ namespace Utils {
...
+const std::string kV1Config("v1");
 
 // The name for the mixer server cluster.
 const std::string kDefaultMixerClusterName("mixer_server");
 
+// ReadConfig() finds config from |json| that matches version |config_version|,
+// and parses config into |message|. Returns true if config is read and parsed
+// successfully.
+bool ReadConfig(const Json::Object &json, Message *message,
+                const std::string &config_version) {
```
Comment (qiwzhang) [inline src/envoy/utils/config.cc]: per style guide,  output argument should be the last one.

File: src/envoy/utils/config.cc [hunk 2]
```diff
@@ -47,19 +68,14 @@ bool ReadV2Config(const Json::Object &json, Message *message) {
...
-  auto &logger = Logger::Registry::getLog(Logger::Id::config);
-  if (status.ok()) {
-    ENVOY_LOG_TO_LOGGER(logger, info, "V2 mixer client config: {}",
-                        message->DebugString());
-    return true;
+  return ReadConfig(json, message, kV2Config);
+}
+
+bool ReadV1Config(const Json::Object &json, Message *message) {
+  if (!json.hasObject(kV1Config)) {
```
Comment (qiwzhang) [inline src/envoy/utils/config.cc]: this checking can be part of ReadConfig() too.

File: src/envoy/utils/config.cc (hunk 1 above)
Comment (JimmyCYJ) [reply src/envoy/utils/config.cc]: Thanks for pointing that out. Fixed.

File: src/envoy/utils/config.cc (hunk 2 above)
Comment (JimmyCYJ) [reply src/envoy/utils/config.cc]: Done.

**Pull request merged**